packages/devtools_app/test/memory/framework/offline_data_test.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app/src/screens/memory/framework/offline_data/offline_data.dart';
+import 'package:devtools_app/src/screens/memory/panes/chart/controller/chart_data.dart';
+import 'package:devtools_app/src/screens/memory/panes/chart/data/primitives.dart';
+import 'package:devtools_app/src/screens/memory/panes/diff/controller/diff_pane_controller.dart';
+import 'package:devtools_app/src/screens/memory/panes/profile/profile_pane_controller.dart';
+import 'package:devtools_app/src/screens/memory/shared/heap/class_filter.dart';
+import 'package:devtools_app/src/screens/memory/shared/primitives/memory_timeline.dart';
+import 'package:devtools_app/src/shared/primitives/simple_items.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test(
+    '$OfflineMemoryData serializes and deserializes correctly',
+    () {
+      final item = OfflineMemoryData(
+        DiffPaneController(loader: null),
+        ProfilePaneController(mode: ControllerCreationMode.connected),
+        ChartData(
+          mode: ControllerCreationMode.offlineData,
+          isDeviceAndroid: true,
+          timeline: MemoryTimeline(),
+          interval: ChartInterval.theDefault,
+          isLegendVisible: true,
+        ),
+        ClassFilter.empty(),
+        selectedTab: 0,
+      );
+
+      final json = item.toJson();
```

Comment:
assert values of toJson result

Reply:
see https://github.com/flutter/devtools/pull/7699#discussion_r1596044223